src/jit/optimizer.cpp

```diff
@@ -6203,6 +6203,12 @@ bool Compiler::optHoistLoopExprsForTree(
         {
             treeIsHoistable = false;
         }
+
+        // Avoid to hoist reference loading
+        if (tree->OperGet() == GT_CLS_VAR && tree->gtType == TYP_BYREF)
+        {
+            treeIsHoistable = treeIsInvariant = false;
+        }
```

Comment:
Just now taking a look this code:

Due to the code block above (line 6189-6194) it would be impossible for treeIsHoistable top be true when `(tree->OperGet() == GT_CLS_VAR)`

What did you observe here?   Is this possibly a native compiler bug?